irods_browser_app/views.py

```diff
@@ -0,0 +1,137 @@
+import json
+import os
+from django.http import HttpResponse, HttpResponseRedirect
+from irods.session import iRODSSession
+from irods.exception import CollectionDoesNotExist
+from hs_core import hydroshare
+
+irods_sess = None
+
+def search_ds(coll):
+    store = {}
+    file = []
+    folder = []
+    if coll.data_objects:
+        for files in coll.data_objects:
+            file.append(files.name)
+    if coll.subcollections:
+        for folders in coll.subcollections:
+            folder.append(folders.name)
+
+    store['files'] = file
+    store['folder'] = folder
+    return store
+
+# Create your views here.
+def login(request):
+    if request.method == 'POST':
+        port = int(request.POST['port'])
+        user = str(request.POST['user'])
+        password = str(request.POST['password'])
+        zone = str(request.POST['zone'])
+        host = str(request.POST['host'])
+        datastore = "/%s/home/%s" % (zone, user)
+
+        response_data = {}
+
+        global irods_sess
+        irods_sess = iRODSSession(user=user, password=password, zone=zone, host=host, port=port)
```

Comment:
If you keep this session object global, it will be shared between all users. Because production can run with more than one process, the same user may first have this login view handled by one process, which will store the global session, and then another process might handle the next request from them using a totally different user's credentials.

The credentials should not be stored, either on disc or in memory, past the lifespan of a single request/response cycle.

Alternatively, perhaps you could send the credentials in each operation that uses them and avoid this login view entirely?